### PR TITLE
chore: upgrade styled components v6 xstyled v4

### PR DESCRIPTION
**Pull request by mleralec**
### Upgrade guide
- Upgrade `styled-components` to v6
- Upgrade `@xstyled/styled-components` to v4
- Remove `@types/styled-components` (types are now included in `styled-components@6.*.*`)
- Use `styled.whateverBox` instead of `styled.div` with `${system}` pattern (The definition of the return type of the styled function has change)
```ts
const StyledTest = styled.div`
  ${system}
`
const Test = () => <Test backgroundColor="tomato" />
```
> This snippet will throw an error because the prop `backgroundColor` doesn't exit on `StyledTest`

To fix the error, you need to use the following pattern:
```ts
const StyledTest = styled.box``
// or
const StyledTest = styled.div<WuiProps>`
  ${system}
` 
```
> Those two snippets are equivalent, please use the first one (more concise)

- Update the rule `toHaveStyleRule` on test by adding spaces on `rgb/rgba` keywords:
```diff
- expect(yourElement).toHaveStyleRule('color', 'rgba(0,0,0,1)')
+ expect(yourElement).toHaveStyleRule('color', 'rgba(0, 0, 0, 1)')
```

### Other breaking changes:
- Removed deprecated stuff on `@welcome-ui/system`
- Some changes on `theme` cf https://github.com/WTTJ/welcome-ui/pull/2524/commits/1736741ee4c1fc6506f1d9c6a9c50d34f5150a79

### CODE REVIEW

File: website/app/blog/page.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ const Home = () => {
   return (
     <Box as="main" p="xl">
       <Box margin="0 auto" maxWidth={600}>
-        <Text color="secondary-red" textAlign="center" variant="subtitle-md">
```
Comment (mleralec) [inline website/app/blog/page.tsx:66]: cc @theo-mesnil what is this color? 🤔

Comment (theo-mesnil) [reply website/app/blog/page.tsx:66]: An invention (it's the next colors but it's not ok here :D)

Comment (theo-mesnil) [reply website/app/blog/page.tsx:66]: @mleralec it's fixed here https://github.com/WTTJ/welcome-ui/pull/2539

**Comment by mleralec** [comment]
I close the pull request because I don't want to see this anymore inside my https://github.com/pulls 🧹 
I keep the branch, feel free to re-open the PR on your side :)

**Pull request closed without merging**